llvm/include/llvm/Transforms/Utils/BasicBlockUtils.h

```diff
@@ -705,6 +705,8 @@ void InvertBranch(BranchInst *PBI, IRBuilderBase &Builder);
 // Check whether the function only has simple terminator:
 // br/brcond/unreachable/ret
 bool hasOnlySimpleTerminator(const Function &F);
+
+bool isPresplitCoroSuspendExit(const BasicBlock &BB);
```

Comment:
should the minimal spanning tree code use this interface too?

Reply:
Good idea - done. It needed a little rework - in the MST case we want to work with edges and block splitting the suspend faux edge. In the coro promotion case, we care about the basic block, and the property that none of the incoming edges is a faux suspend edge.